### PR TITLE
Add JSON config support

### PR DESCRIPTION
**Pull request by slimepointe**
default values are already inputted into the json file

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -291,13 +292,23 @@ async def main(args):
...
-  parser.add_argument("--host", default="127.0.0.1", help="The hostname the server will listen on.")
-  parser.add_argument("--port", default=6001, help="The TCP port the server will listen on.")
-  parser.add_argument("--static", help="Where static files are served from.")
-  parser.add_argument("--limits", action="store_true", help="Enable rate limits.")
-  parser.add_argument("--bandwidth", default=1000, help="Bandwidth limit per IP, in kilobytes per second.")
-  parser.add_argument("--connections", default=30, help="Connections limit per IP, in kilobytes per second.")
-  parser.add_argument("--window", default=60, help="Fixed window length for rate limits, in seconds.")
-  args = parser.parse_args()
+  def load_config():
+    with open('config.json') as f:
```
Comment (ading2210) [inline main.py:296]: Why is the path hardcoded?

File: main.py
Comment (velzie) [inline main.py:1]: why are you removing command line options

File: main.py [hunk 2]
```diff
@@ -291,13 +292,23 @@ async def main(args):
...
+    host=config["host"],
+    port=config["port"],
+    static=config["static"],
+    limits=config["limits"],
+    bandwidth=config["bandwidth"],
+    connections=config["connections"],
+    window=config["window"]
+  )
+
+  print(args)
```
Comment (ading2210) [inline main.py]: Why is there a hardcoded print statement here?

File: main.py
Comment (slimepointe) [reply main.py:1]: i assumed that using a json config would remove the command line arguments, that's just how i took the roadmap

File: main.py [hunk 3]
```diff
@@ -291,13 +292,23 @@ async def main(args):
...
   )
 
-  parser.add_argument("--host", default="127.0.0.1", help="The hostname the server will listen on.")
-  parser.add_argument("--port", default=6001, help="The TCP port the server will listen on.")
-  parser.add_argument("--static", help="Where static files are served from.")
-  parser.add_argument("--limits", action="store_true", help="Enable rate limits.")
-  parser.add_argument("--bandwidth", default=1000, help="Bandwidth limit per IP, in kilobytes per second.")
-  parser.add_argument("--connections", default=30, help="Connections limit per IP, in kilobytes per second.")
-  parser.add_argument("--window", default=60, help="Fixed window length for rate limits, in seconds.")
-  args = parser.parse_args()
```
Comment (ading2210) [inline main.py:301]: There is no reason to remove the CLI arguments.

File: main.py (hunk 1 above)
Comment (slimepointe) [reply main.py:296]: do you want me to take the config file as an input? i dont see any reason why it wouldnt be hardcoded when it is included in the repo

Comment (ading2210) [reply main.py:296]: Ideally loading the config file should be done through a `--config` option or something.

Comment (slimepointe) [reply main.py:296]: i can add it using sys.argv if you would like that then i'll reopen the pr?

File: main.py (hunk 2 above)
Comment (slimepointe) [reply main.py]: i fixed that, that was my bad for debugging purposes

File: main.py (hunk 3 above)
Comment (slimepointe) [reply main.py:301]: that's how i took the roadmap, remove cli arguments to use a json config

File: main.py (hunk 1 above)
Comment (ading2210) [reply main.py:296]: Don't use `sys.argv` when the argsparse module exists and is already used in the repo.

**Pull request closed without merging**